pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/AggregationOperator.java

```diff
@@ -44,29 +44,28 @@
   private List<AggregationInfo> _aggregationInfoList;
   private MProjectionOperator _projectionOperator;
 
-  private IndexSegment _indexSegment;
   private int _nextBlockCallCounter = 0;
 
   private int[] _reusableDocIdSet;
 
+  private final long _totalRawDocs;
+
   /**
    * Constructor for the class.
    *
    * @param indexSegment Index on which aggregation group by is to be performed.
    * @param aggregationsInfoList List of AggregationInfo (contains context for applying aggregation functions).
    * @param projectionOperator Projection
    */
-  public AggregationOperator(IndexSegment indexSegment, List<AggregationInfo> aggregationsInfoList,
-      MProjectionOperator projectionOperator) {
+  public AggregationOperator(List<AggregationInfo> aggregationsInfoList, MProjectionOperator projectionOperator, long totalRawDocs) {
```

Comment:
Although I don't have a better alternative to suggest, passing totalRawDocs seems a bit odd. Would it make more sense to propagate it via block interface (bottom up)?